runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java

```diff
@@ -233,12 +233,11 @@ public void gc(long trimMark) {
             if (((ICorfuSMR) obj).getCorfuSMRProxy() instanceof CorfuCompileProxy) {
                 ((CorfuCompileProxy) ((ICorfuSMR) obj).
                         getCorfuSMRProxy()).getUnderlyingObject().gc(trimMark);
+            } else {
+                // MVOCorfuCompileProxy
+                ((MultiVersionObject) obj).gc(trimMark);
             }
```

Comment:
It looks like there's a typecast error here when casting obj to MultiVersionObject. Also, with the NO_CACHE option set, the address space metadata for the MVO will not get cleaned up.

Reply:
The idea of `NO_CACHE` objects is to drop the object as a whole when the object is no longer used, whereas other `CACHE` objects will remain in ObjectsView cache. This is the same to VLO objects - they are not gc-ed by ViewsGarbageCollector.

Reply:
I just pushed a new change to correct the type cast. Thanks!